superset/connectors/sqla/models.py

```diff
@@ -536,14 +540,15 @@ def query(self, query_obj):
         qry_start_dttm = datetime.now()
         engine = self.database.get_sqla_engine()
         qry = self.get_sqla_query(**query_obj)
-        sql = str(qry)
+        sql = self.get_query_str(**query_obj)
```

Comment:
This is the actual fix, the rest of the PR is mostly syntactic improvements I touched up while debugging.